cpp/src/join/hash_join.cu

```diff
@@ -375,8 +375,9 @@ hash_join<Hasher>::hash_join(cudf::table_view const& build,
 {
   CUDF_FUNC_RANGE();
   CUDF_EXPECTS(0 != build.num_columns(), "Hash join build table is empty");
-  CUDF_EXPECTS(build.num_rows() < cudf::detail::MAX_JOIN_SIZE,
-               "Build column size is too big for hash join");
+  CUDF_EXPECTS(build.num_rows() <= cudf::detail::MAX_JOIN_SIZE,
```

Comment:
Do comparisons to the `size_type` max always need to be done as type `std::size_t`? I don't see how this could ever be false, especially if both operands are `size_type`.

Reply:
We're comparing a `size_type` to the value `constexpr size_type MAX_JOIN_SIZE{std::numeric_limits<size_type>::max()};`. This can never be false, so this `EXPECTS` cannot raise.

Reply:
You're right. This makes no sense and we should be able to remove `MAX_JOIN_SIZE` completely. I traced the history and this constant has existed for a very long time. It is possible it was intended to be smaller (or maybe at least different) than `size_type` at one time -- perhaps when prototyping. And the value has not changed since it was added in 2019. So I think we can safely remove it now.